purchases/src/main/kotlin/com/revenuecat/purchases/customercenter/CustomerCenterConfigData.kt

```diff
@@ -0,0 +1,465 @@
+package com.revenuecat.purchases.customercenter
+
+import com.revenuecat.purchases.paywalls.PaywallColor
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+typealias RCColor = PaywallColor
+
+private const val HASH_CODE_MULTIPLIER = 31
+
+@Serializable
+internal class CustomerCenterRoot(
+    @SerialName("customer_center") val customerCenter: CustomerCenterConfigData,
+)
+
+@Serializable
+class CustomerCenterConfigData(
+    @Serializable(with = ScreenMapSerializer::class) val screens: Map<Screen.ScreenType, Screen>,
+    val appearance: Appearance,
+    val localization: Localization,
+    val support: Support,
+    @SerialName("last_published_app_version") val lastPublishedAppVersion: String? = null,
+) {
+    @Serializable
+    class Localization(
+        val locale: String,
+        @SerialName("localized_strings") val localizedStrings: Map<String, String>,
```

Comment:
I guess if we add a default map here containing every enum with its default value, it will be entirely replaced if the response contains a `localized_strings` map, right? Would be great if the maps would be merged, giving the map in the response precedence.

Reply:
>  it will be entirely replaced if the response contains a localized_strings map, right?

That's correct.

> Would be great if the maps would be merged, giving the map in the response precedence.

Yeah that would be good... Maybe we can write a custom serializer to do that...

Reply:
Actually, after thinking about it, I think it might be this way to be able to access strings that are not part of the enum? I don't see a reason why we should allow that, but it's something that would be limited if we make this map a map of `CommonLocalizedString` to `String`... I think I might leave it as is to keep the same API as in iOS. cc @vegaro

Reply:
hmmm, that's a good point, at the end of the day these are just common strings for now by definition. We could at some point return more arbitrary strings that are arbitrary, and don't have a default. I can't think of a usage for that now but might be better to keep that flexibility. The cool thing about making it `CommonLocalizedString` to `String` is that the map will only contain strings that are used by that version of the SDK.

I think it's fine to just do what we do in iOS already.

Reply:
I'm fine with doing what we do in iOS, yes. Let's not write a custom serializer just for this. Was just musing 😇